defs.sh

```diff
@@ -27,22 +27,29 @@ then
     exit 1
 fi
 
-# Versions of things (do this before config.sh so they can be config'd)
-BINUTILS_URL=http://ftpmirror.gnu.org/gnu/binutils/binutils-2.25.1.tar.bz2
-#BINUTILS_URL=http://mirrors.kernel.org/sourceware/binutils/snapshots/binutils-2.24.90.tar.bz2
+# Default HTTP/HTTPS protocol
+test -z "$HTTP_PROTO" && HTTP_PROTO=https
```

Comment:
i think this needs to come after sourcing the config...

Reply:
can you show me the line that sources config.sh before this statement ? afaict the first invocation is on line 32

Reply:
Oh, I'm sorry, I was looking at the wrong line. I'll fix it.

Reply:
Oh, I was looking at the wrong line, sorry. Lemme fix it.

Reply:
Wait, no, this is correct. If the config file defines HTTP_PROTO, then that'll just override the default set here, because the config is sourced afterwards.

Reply:
test -z checks whether the variable is unset. this line is meant to be used after including config.sh.

Reply:
> test -z checks whether the variable is unset.

Right, I know. If the environment variable is unset or set to the empty string (that's why I checked the string was zero-length and [not that it was unset](https://stackoverflow.com/a/13864829/1198896)), then the test will pass and set the default value of `https`.

But if it's set in the config file then the config file will override, regardless of what the variable has been set to already (either the default or to whatever's in the environment).

I mean, I suppose the behavior is the same if it's moved after `. config.sh`? So I can move it if you want